types/sanctuary/index.d.ts

```diff
@@ -107,19 +103,25 @@ declare namespace Sanctuary {
     gte<A>(x: Ord<A>): (y: Ord<A>) => boolean;
     min<A>(x: Ord<A>): (y: Ord<A>) => A;
     max<A>(x: Ord<A>): (y: Ord<A>) => A;
+    clamp(tk: TK): (tk: TK) => (tk: TK) => TK;
     id<A>(p: TypeRep): Fn<A, A> | Category<any>;
     concat<A>(x: Semigroup<A>): (y: Semigroup<A>) => Semigroup<A>;
-    concat<A>(x: ReadonlyArray<A>): (y: ReadonlyArray<A>) => A[];
+    concat<A>(x: ReadonlyArray<A>): (y: ReadonlyArray<A>) => Array<A>;
     concat<A>(x: StrMap<A>): (y: StrMap<A>) => StrMap<A>;
     concat(x: string): (y: string) => string;
     empty(p: TypeRep): Monoid<any>;
+    invert(tk: TK): TK;
+    filter(tk: TK): (tk: TK) => TK;
+    reject(tk: TK): (tk: TK) => TK;
     map<A, B>(p: Fn<A, B>): {
       <C>(q: Fn<C, A>): Fn<C, B>;
-        (q: ReadonlyArray<A>): B[];
+        (q: ReadonlyArray<A>): Array<B>;
```

Comment:
For simple types please use regular `T[]`

Reply:
I favour `Array<T>` for three reasons:

  - I dislike special cases;
  - I find `T[][]` and `Maybe<T[]>` more difficult to parse than `Array<Array<T>>` and `Maybe<Array<T>>`; and
  - it is closer to the Hindley–Milner representation, the canonical representation in the Sanctuary ecosystem.

What are your reasons for favouring `T[]`, @vicrac?

Reply:
It's shorter, more natural and is favoured by ts-linter by default. Can't think of any reason why parsing is a matter?

Reply:
> Can't think of any reason why parsing is a matter?

I meant that it's easier for me to parse as a human. There is no syntactic nesting in `T[][]`, whereas the syntactic nesting of `Array<Array<T>>` corresponds to the semantic nesting.